SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMapGeneration.cpp

```diff
@@ -52,22 +54,36 @@ bool GenerateTestMaps()
 	TArray<UClass*> TestMapClasses;
 	for (TObjectIterator<UClass> Iter; Iter; ++Iter)
 	{
-		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) && *Iter != UGeneratedTestMap::StaticClass())
+		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) || Iter->IsChildOf(ASpatialFunctionalTest::StaticClass()))
 		{
 			TestMapClasses.Add(*Iter);
 		}
 	}
 
 	for (UClass* TestMapClass : TestMapClasses)
 	{
-		UGeneratedTestMap* TestMap = NewObject<UGeneratedTestMap>(GetTransientPackage(), TestMapClass);
-		TestMap->AddToRoot(); // Okay, must admit, not completely sure what's going on here, seems like even though the commandlet is
-							  // the outer of the newly generated object, the object still gets GCed when creating a new map, so have to add
-							  // to root here to prevent GC
+		IGeneratableTestMap* TestMap;
+		if (TestMapClass->IsChildOf(ASpatialFunctionalTest::StaticClass()))
+		{
+			TestMap = NewObject<ASpatialFunctionalTest>(GetTransientPackage(), TestMapClass);
+		}
+		else
+		{
+			TestMap = NewObject<UGeneratedTestMap>(GetTransientPackage(), TestMapClass);
+		}
+
+		UObject* TestMapObjPtr = CastChecked<UObject>(TestMap);
```

Comment:
just learnt about CastChecked, very cool, might be able to use it in a few places

Reply:
Note that it will assert and stop the editor, so use carefully - it is better for  test code (and editor code in general) to alert to problem and abort (e.g. fail a test)

Reply:
It will run as part of the editor flow, maps are not generated when the test is run but when the editor is started or when the generate map button in the editor is pressed.

Reply:
I wasn't clear, I was talking generally - both "editor flow" and "test flow" code should warn rather than assert - perhaps the example I gave should have matched the context on this code (and not talk about tests).